### PR TITLE
fix(reconnect): move not-authorized retry to XmppConnection

### PR DESCRIPTION
**Pull request by virtuacoplenny**

### CODE REVIEW

File: spot-client/src/common/remote-control/BaseRemoteControlService.js [hunk 1]
```diff
@@ -151,7 +153,7 @@ export class BaseRemoteControlService extends Emitter {
         // wait a little bit to retry to avoid a stampeding herd
         const jitter = getJitterDelay();
 
-        const previousJoinCode = this.getJoinCode();
+        this._previousJoinCode = this.getJoinCode() || this._previousJoinCode;
```
Comment (paweldomas) [inline spot-client/src/common/remote-control/BaseRemoteControlService.js]: why is this change necessary ?

Comment (virtuacoplenny) [reply spot-client/src/common/remote-control/BaseRemoteControlService.js]: getJoinCode retrieves the muc + lock from the xmpp connection. If the connection does not exist an empty string is returned. On the first reconnect attempt this is fine. On the second reconect attempt, the xmpp connection has already been destroyed so getJoinCode will return an empty string. Would you prefer caching of the last join code instead of getJoinCode retrieving from the xmpp connection each time?

Comment (virtuacoplenny) [reply spot-client/src/common/remote-control/BaseRemoteControlService.js]: Related question: should disconnect() ever be called during the reconnect flow?

Comment (paweldomas) [reply spot-client/src/common/remote-control/BaseRemoteControlService.js]: >getJoinCode retrieves the muc + lock from the xmpp connection. If the connection does not exist an empty string is returned. On the first reconnect attempt this is fine. On the second reconect attempt, the xmpp connection has already been destroyed so getJoinCode will return an empty string. Would you prefer caching of the last join code instead of getJoinCode retrieving from the xmpp connection each time?

I think I have this change or was planning to do as part of the backend PR, so it is good for now.

> Related question: should disconnect() ever be called during the reconnect flow?

If it would emit disconnected events the it shouldn't. I'd say whatever is needed by the XMPP connection to re-establish the connection.
